### PR TITLE
pkp/pkp-lib#7863 added submission decision api end point

### PR DESCRIPTION
**Pull request by touhidurabir**
This PR aims to provide solution for pkp/pkp-lib#7863

### CODE REVIEW

File: api/v1/submissions/PKPSubmissionHandler.php [hunk 1]
```diff
@@ -550,6 +556,40 @@ public function delete($slimRequest, $response, $args)
...
     }
 
+    /**
+     * Get the decisions associated to a submission
+     *
+     * @param Request $slimRequest Slim request object
+     * @param Response $response object
+     * @param array $args arguments
+     *
+     * @return Response
```
Comment (NateWr) [inline api/v1/submissions/PKPSubmissionHandler.php:566]: I know the existing handlers use the `@param` and `@return` a lot, but let's try to start type-hinting in the function signature.

File: api/v1/submissions/PKPSubmissionHandler.php [hunk 2]
```diff
@@ -550,6 +556,40 @@ public function delete($slimRequest, $response, $args)
...
+
+        $decisionIterator = Repo::decision()->getCollector()
+            ->filterBySubmissionIds([$submission->getId()])
+            ->getMany();
+        
+        $map = Repo::decision()->getSchemaMap();
+
+        foreach ($decisionIterator as $decision) {
+            $data[] = $map->map($decision);
+        }
```
Comment (NateWr) [inline api/v1/submissions/PKPSubmissionHandler.php:588]: There should be a `mapMany()` function that you can use: `Repo::decision()->getSchemaMap()->mapMany($decisionIterator)`. When you pass the JSON data back, you'll want ot reset the array keys so:

```php
$data = Repo::decision()
    ->getSchemaMap()
    ->mapMany($decisionIterator->values());
```

File: api/v1/submissions/PKPSubmissionHandler.php [hunk 3]
```diff
@@ -550,6 +556,40 @@ public function delete($slimRequest, $response, $args)
         return $response->withJson($submissionProps, 200);
     }
 
+    /**
+     * Get the decisions associated to a submission
```
[the remark this answers is not in the document]
Comment (touhidurabir) [reply api/v1/submissions/PKPSubmissionHandler.php]: Done .

File: api/v1/submissions/PKPSubmissionHandler.php (hunk 1 above)
Comment (touhidurabir) [reply api/v1/submissions/PKPSubmissionHandler.php]: Done.

File: api/v1/submissions/PKPSubmissionHandler.php (hunk 2 above)
Comment (touhidurabir) [reply api/v1/submissions/PKPSubmissionHandler.php]: Done.

**Pull request merged**